dcc-portal-server/src/main/java/org/icgc/dcc/portal/server/service/FileService.java

```diff
@@ -358,10 +360,12 @@ private static String toAverageSizeString(SearchHitField hitField) {
     return result;
   }
 
-  private static List<File> convertHitsToRepoFiles(SearchHits hits) {
+  private static List<File> convertHitsToRepoFiles(SearchHits hits, Query query) {
     return FluentIterable.from(hits)
-        .transform(hit -> parse(hit.getSourceAsString()))
+        .transform(hit -> {
+          Map<String, Object> fieldMap = createResponseMap(hit, query, EntityType.FILE);
+          return parse(fieldMap);
```

Comment:
``java
return Stream.of(hits)
  .map(hit -> createResponseMap(hit, query, EntityType.FILE))
  .map(this::parse)
  .collect(toImmutlableList())
```